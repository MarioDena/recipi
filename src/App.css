body {
  background-color: white;
  background-image:
    linear-gradient(
      90deg,
      rgba(200, 0, 0, 0.3) 50%,
      transparent 20%
    ),
    linear-gradient(rgba(200, 0, 0, 0.3) 50%, transparent 20%);
  background-size: 50px 50px;
}

.loaderContainer {
  height: calc(100vh - 220px);
}

.card {
  max-width: 300px;
  height: 450px;
  margin: 10px;
}

.goback {
  margin-top: 40px;
  margin-right: 20px;
  margin-bottom: 80px;
}

.instructions3 {
  margin-right: 20px;
}

.instruction2 {
  margin-top: 20px;
  padding-top: 20px;
}

.instructions {
  margin-left: 10px;
  margin-right: 10px;
}

.img-fluid {
  width: 100%;
}

.catTitle {
  width: 100%;
  display: flex;
  flex-direction: column;
  background-color: white;
  text-align: center;
}

.loading {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.FullMealcontainer {
  width: 100%;
  display: flex;
  font-family: 'Lato', sans-serif;
}

.my-3 {
  margin-top: 0 !important;
}

.catContent {
  font-size: 50px;
  font-family: 'Pacifico', cursive;
  text-decoration: none;
  color: rgb(4, 196, 255);
  margin-bottom: 50px;
}

.sticky-nav {
  position: sticky;
  top: 0;
  z-index: 200;
}

.card-title {
  font-family: 'Lato', sans-serif;
}

.centered {
  background-color: white;
  display: flex;
  max-width: 1100px;
}

.card-body {
  opacity: 1;
}

.card-img-top {
  cursor: pointer;
}

.filter {
  width: 100%;
  height: 80px;
}

.card-group {
  margin: auto;
  width: 960px;
}

.card-group::after {
  content: '';
  flex: auto;
}

.cook-btn {
  position: absolute;
  right: 10px;
  bottom: 10px;
  font-weight: bold;
}

.Coco {
  font-size: 50px;
  font-family: 'Pacifico', cursive;
  text-decoration: none;
  color: rgb(4, 196, 255);
}

.Coco:hover {
  text-decoration: none;
}

.Coco:active {
  text-decoration: none;
}

.Coco:visited {
  color: rgb(4, 196, 255);
}

@media only screen and (max-width: 1199px) {
  .card-group {
    width: 650px;
  }

  .centered {
    background-color: white;
    display: flex;
    align-items: center;
    max-width: 670px;
  }

  .my-3 {
    margin-top: -10px !important;
  }
}

@media only screen and (max-width: 768px) {
  .card-group {
    max-width: 620px;
  }

  .my-3 {
    margin-top: 10px !important;
  }

  .centered {
    background-color: white;
    display: flex;
    align-items: center;
    max-width: 600px;
  }

  .card {
    max-width: 600px;
    height: auto;
    margin: 10px;
  }

  .card-title {
    max-width: 250px;
  }
}

@media only screen and (max-width: 400px) {
  .card-title {
    max-width: 150px;
    margin-top: -10px;
    margin-left: -10px;
  }

  .catContent {
    font-size: 30px;
    margin-bottom: 30px;
  }
}

@media only screen and (max-width: 350px) {
  .Coco {
    font-size: 22px;
  }

  .filter {
    width: 100%;
    height: 40px;
  }

  .card-title {
    font-size: 16px;
    max-width: 100px;
  }
}

@media only screen and (max-width: 240px) {
  .card-title {
    font-size: 12px;
    max-width: 60px;
  }

  .btn-danger {
    font-size: 12px;
  }

  .cook-btn {
    width: 50px;
    height: 30px;
    font-size: 10px;
    line-height: 50%;
  }

  .Coco {
    font-size: 16px;
  }
}
