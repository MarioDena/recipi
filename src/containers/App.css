body {
  background-color: black;
}

.App {
  text-align: center;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.imageGal:hover {
  padding: 10px;
}

.custmText {
  font-family: 'Pacifico', cursive;
}

.dropdown-content li > a {
  color: white;
}

.dropdown-content li > a:hover {
  color: black;
}

.mealcard {
  cursor: pointer;
  position: relative;
}

.catContent {
  font-family: 'Pacifico', cursive;
  font-size: 50px;
  padding-top: 40px;
  min-height: 150px;
}

.findbtn {
  margin-top: 40px;
  margin-bottom: 40px;
  background-color: white;
  color: #282c34;
}

.findbtn:hover {
  background-color: #282c34;
  color: white;
}

.my-3 {
  font-family: 'Pacifico', cursive;
}

.smallCont {
  text-align: center;
}

.instructions {
  max-width: 900px;
}

.instruction2 {
  font-family: 'Pacifico', cursive;
  text-align: center;
}

.sidenav-overlay {
  z-index: 600;
}

.img-fluid {
  padding: 50px;
  border-radius: 50%;
}

.ingredients {
  height: 100%;
}
.overlay {
  -webkit-user-select: none; /* Safari */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* IE10+/Edge */
  user-select: none; /* Standard */
  cursor: pointer;
  font-family: 'Pacifico', cursive;
  font-size: 20px;
  width: calc(100% - 40px);
  position: absolute;
  text-align: center;
  padding-top: 5px;
  height: 60px;
  left: 20px;
  bottom: 10px;
  background-color: rgba(0, 0, 0, 0.651);
}

.dropdown-trigger:hover {
  background-color: rgb(228, 228, 228);
  color: black;
}

nav {
  padding-bottom: -10px;
  border-bottom: 1px solid white;
}

.imageGal {
  border-radius: 5px;
  padding: 20px;
}

img {
  width: 100%;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}
.brand-logo {
  font-size: 50px;
  font-family: 'Pacifico', cursive;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media only screen and (min-width: 1200px) {
  .forceCenter {
    padding-left: 100px;
    padding-right: 100px;
  }
}

@media only screen and (min-width: 1500px) {
  .forceCenter {
    padding-left: 200px;
    padding-right: 200px;
  }
}

@media only screen and (max-width: 1450px) {
  .overlay {
    font-size: 16px;
  }
}

@media only screen and (max-width: 1160px) {
  .overlay {
    font-size: 14px;
  }
}

@media only screen and (max-width: 1070px) {
  .overlay {
    font-size: 12px;
  }
}

@media only screen and (max-width: 992px) {
  .overlay {
    font-size: 20px;
  }
}

@media only screen and (max-width: 800px) {
  .overlay {
    font-size: 16px;
  }
}

@media only screen and (max-width: 640px) {
  .overlay {
    font-size: 14px;
  }
}

@media only screen and (max-width: 600px) {
  .overlay {
    font-size: 20px;
  }
}

@media only screen and (max-width: 340px) {
  .overlay {
    font-size: 14px;
  }
}
