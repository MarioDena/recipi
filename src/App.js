import React from 'react';
import MealList from './containers/mealList';
import './App.css';

function App() {
  return <MealList />;
}

export default App;
