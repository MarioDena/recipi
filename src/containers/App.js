import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { filterMeals } from '../actions/index';
import CategoryFilter from '../components/navbar';
import MealList from '../components/mealList';
import 'materialize-css/dist/css/materialize.min.css';
import './App.css';

const mapDispatchToProps = (dispatch) => {
  return {
    changeCategory: (filter) => dispatch(filterMeals(filter)),
  };
};

const mapStateToProps = (state) => {
  return {
    category: state.filter,
  };
};

const App = (props) => {
  const { changeCategory } = props;
  const switchFilter = (filter) => {
    changeCategory(filter);
  };
  return (
    <BrowserRouter>
      <CategoryFilter changeFilter={switchFilter} />
      <Route exact path="/" component={MealList} />
    </BrowserRouter>
  );
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
