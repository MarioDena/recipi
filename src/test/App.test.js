import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';
import App from '../containers/App';
import CategoryFilter from '../components/navbar';
import rootReducer from '../reducers/index';

const store = createStore(rootReducer);

afterEach(cleanup);

test('Renders the app enterely', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Contains the navbar', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Provider store={store}>
        <CategoryFilter />
      </Provider>
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
