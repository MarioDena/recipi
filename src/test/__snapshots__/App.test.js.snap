// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contains the navbar 1`] = `
<DocumentFragment>
  <div
    class="navbar-fixed"
  >
    <nav
      class="black"
    >
      <ul
        class="dropdown-content black"
        id="dropdown1"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Breakfast"
          >
            Breakfast
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Dessert"
          >
            Dessert
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Miscellaneous"
          >
            Miscellaneous
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Pasta"
          >
            Pasta
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Side"
          >
            Side
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Starter"
          >
            Starter
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Vegetarian"
          >
            Vegetarian
          </a>
        </li>
      </ul>
      <ul
        class="sidenav"
        id="mobile-demo"
      >
        <li>
          <div>
            <h5
              class="custmText center-align grey-text text-darken-4"
            >
              Categories
            </h5>
          </div>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Breakfast"
          >
            Breakfast
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Dessert"
          >
            Dessert
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Miscellaneous"
          >
            Miscellaneous
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Pasta"
          >
            Pasta
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Side"
          >
            Side
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Starter"
          >
            Starter
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Vegetarian"
          >
            Vegetarian
          </a>
        </li>
      </ul>
      <div
        class="container"
      >
        <div
          class="nav-wrapper"
        >
          <a
            class="brand-logo"
            href="/"
          >
            Recipi
          </a>
          <a
            aria-current="page"
            class="sidenav-trigger active"
            data-target="mobile-demo"
            href="/"
          >
            <i
              class="material-icons"
            >
              menu
            </i>
          </a>
          <ul
            class="right hide-on-med-and-down"
          >
            <li>
              <a
                class="dropdown-trigger"
                data-target="dropdown1"
                href="#"
              >
                Categories
                <i
                  class="material-icons right"
                >
                  arrow_drop_down
                </i>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Renders the app enterely 1`] = `
<DocumentFragment>
  <div
    class="navbar-fixed"
  >
    <nav
      class="black"
    >
      <ul
        class="dropdown-content black"
        id="dropdown1"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Breakfast"
          >
            Breakfast
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Dessert"
          >
            Dessert
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Miscellaneous"
          >
            Miscellaneous
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Pasta"
          >
            Pasta
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Side"
          >
            Side
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Starter"
          >
            Starter
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            title="Vegetarian"
          >
            Vegetarian
          </a>
        </li>
      </ul>
      <ul
        class="sidenav"
        id="mobile-demo"
      >
        <li>
          <div>
            <h5
              class="custmText center-align grey-text text-darken-4"
            >
              Categories
            </h5>
          </div>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Breakfast"
          >
            Breakfast
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Dessert"
          >
            Dessert
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Miscellaneous"
          >
            Miscellaneous
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Pasta"
          >
            Pasta
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Side"
          >
            Side
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Starter"
          >
            Starter
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="sidenav-close grey-text text-darken-4 active"
            href="/"
            title="Vegetarian"
          >
            Vegetarian
          </a>
        </li>
      </ul>
      <div
        class="container"
      >
        <div
          class="nav-wrapper"
        >
          <a
            class="brand-logo"
            href="/"
          >
            Recipi
          </a>
          <a
            aria-current="page"
            class="sidenav-trigger active"
            data-target="mobile-demo"
            href="/"
          >
            <i
              class="material-icons"
            >
              menu
            </i>
          </a>
          <ul
            class="right hide-on-med-and-down"
          >
            <li>
              <a
                class="dropdown-trigger"
                data-target="dropdown1"
                href="#"
              >
                Categories
                <i
                  class="material-icons right"
                >
                  arrow_drop_down
                </i>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div>
    <div
      class="container"
    >
      Loading...
    </div>
  </div>
</DocumentFragment>
`;
